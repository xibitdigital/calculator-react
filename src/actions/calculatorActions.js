import * as Types from '../constants'

export const clear = () => ({ type: Types.CLEAR })

export const enterNumber = number => ({ number, type: Types.ENTER_NUMBER })

export const evaluate = () => ({ type: Types.EVALUATE })

export const percentage = () => ({ type: Types.PERCENTAGE })

export const setOperator = operator => ({ operator, type: Types.SET_OPERATOR })

export const toggleNegative = () => ({ type: Types.TOGGLE_NEGATIVE })
