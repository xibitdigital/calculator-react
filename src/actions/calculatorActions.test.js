import * as calculatorActions from './calculatorActions'

test('should trigger a clear action', () => {
    expect(calculatorActions.clear()).toMatchSnapshot()
})

test('should trigger an enterNumber action', () => {
    expect(calculatorActions.enterNumber(10)).toMatchSnapshot()
})

test('should trigger an evaluate action', () => {
    expect(calculatorActions.evaluate()).toMatchSnapshot()
})

test('should trigger a percentage action', () => {
    expect(calculatorActions.percentage()).toMatchSnapshot()
})

test('should trigger a setOperator action', () => {
    const fn = jest.fn()
    expect(calculatorActions.setOperator(fn)).toMatchSnapshot()
})

test('should trigger a toggleNegative action', () => {
    expect(calculatorActions.toggleNegative()).toMatchSnapshot()
})
