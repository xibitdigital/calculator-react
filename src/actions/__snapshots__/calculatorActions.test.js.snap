// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should trigger a clear action 1`] = `
Object {
  "type": "calculator/CLEAR",
}
`;

exports[`should trigger a percentage action 1`] = `
Object {
  "type": "calculator/PERCENTAGE",
}
`;

exports[`should trigger a setOperator action 1`] = `
Object {
  "operator": [Function],
  "type": "calculator/SET_OPERATOR",
}
`;

exports[`should trigger a toggleNegative action 1`] = `
Object {
  "type": "calculator/TOGGLE_NEGATIVE",
}
`;

exports[`should trigger an enterNumber action 1`] = `
Object {
  "number": 10,
  "type": "calculator/ENTER_NUMBER",
}
`;

exports[`should trigger an evaluate action 1`] = `
Object {
  "type": "calculator/EVALUATE",
}
`;
